### DISCUSSION TITLE
How to handle large ECG dataset?

### DISCUSSION THREAD

**Discussion by BaiJonathan** [neuropsychology/NeuroKit#994] [Q&A]
We  get the 24 hours raw ECG data for the devices(250Hz).  I want to analyze those data and produce HRV analysis.
my code snippet like this:
"
signal = pd.read_csv(filePath)
processed_data, info = nk.bio_process(signal["ECG"], sampling_rate=250)
results = nk.bio_analyze(processed_data, sampling_rate=frequency)
"
NO response.

but maybe it's too big.  I can't get the result even after 30 mins.
so any recommendations ?    how to handle large dataset?  
Thanks in advance.

**Comment by BaiJonathan** [comment]
https://akt.obs.cn-east-3.myhuaweicloud.com/1699942534081277952.csv.zip

the dataset.

**Reply by paudel54** [reply]
Create 1-hour chunks and run for 24 loops. Append the data to the empty list or some database
